domain/schema/secret.go

```diff
@@ -0,0 +1,282 @@
+        pending_delete BOOLEAN NOT NULL DEFAULT (FALSE),
+        expire_time DATETIME,
+        create_time DATETIME NOT NULL DEFAULT (STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW', 'utc')),
+        update_time DATETIME NOT NULL DEFAULT (STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW', 'utc')),
+        CONSTRAINT pk_secret_revision_metadata_uuid_revision
+            PRIMARY KEY (metadata_uuid, revision),
+        CONSTRAINT fk_secret_revision_secret_metadata_uuid 
+            FOREIGN KEY (metadata_uuid)
+            REFERENCES secret_metadata (uuid),
+        CONSTRAINT fk_secret_revision_secret_value_ref_uuid
+            FOREIGN KEY (value_ref_uuid)
+            REFERENCES secret_value_ref (uuid),
+        CONSTRAINT secret_revision_data_or_value_ref_not_null
+            CHECK (
+                (data IS NOT NULL AND value_ref_uuid IS NULL) OR
+                (data IS NULL AND value_ref_uuid IS NOT NULL)
+            )
+    );
+
+CREATE TABLE
+    secret_application_consumed (
+        secret_metadata_uuid TEXT NOT NULL,
+        application_uuid TEXT NOT NULL,
+        label TEXT,
+        current_revision INT NOT NULL,
+        -- TODO:latest_revision was for the consumer watcher can be 
+        -- triggered when a new secret revision is added.
+        -- ????
+        latest_revision INT NOT NULL,
+        CONSTRAINT pk_secret_application_consumed_secret_metadata_uuid_application_uuid
+            PRIMARY KEY (secret_metadata_uuid, application_uuid),
+        CONSTRAINT fk_secret_application_consumed_secret_metadata_uuid
+            FOREIGN KEY (secret_metadata_uuid)
+            REFERENCES secret_metadata (uuid),
+        CONSTRAINT fk_secret_application_consumed_application_uuid
+            FOREIGN KEY (application_uuid)
+            REFERENCES application (uuid)
+    );
+
+-- TODO: same here, we need to ensure the label is unique for the application and all its units!!!
+CREATE UNIQUE INDEX idx_secret_application_consumed_label ON secret_application_consumed (label,application_uuid);
+
+CREATE TABLE
+    secret_unit_consumed (
+        secret_metadata_uuid TEXT NOT NULL,
+        unit_uuid TEXT NOT NULL,
+        label TEXT,
+        current_revision INT NOT NULL,
+        latest_revision INT NOT NULL,
+        CONSTRAINT pk_secret_unit_consumed_secret_metadata_uuid_unit_uuid
+            PRIMARY KEY (secret_metadata_uuid, unit_uuid),
+        CONSTRAINT fk_secret_unit_consumed_secret_metadata_uuid
+            FOREIGN KEY (secret_metadata_uuid)
+            REFERENCES secret_metadata (uuid),
+        CONSTRAINT fk_secret_unit_consumed_unit_uuid
+            FOREIGN KEY (unit_uuid)
+            REFERENCES unit (uuid)
+    );
+
+-- TODO: same here, we need to ensure the label is unique for the application
+-- and all its units ! ! !
+CREATE UNIQUE INDEX idx_secret_unit_consumed_label ON secret_unit_consumed (label,unit_uuid);
+
+-- -- TODO: do we still need this table? If we do, we may need 2 new tables:
```

Comment:
This is used, so we do need it. It's used to help track obsolete revisions for secrets consumed in other models for example.